fs/pty.c

```diff
@@ -51,6 +69,13 @@ static int pty_slave_open(struct tty *tty) {
     return 0;
 }
 
+static int pty_slave_close(struct tty *tty) {
+    if (tty->refcount - 1 == (tty->session ? 2 : 1)) {
```

Comment:
This looks like a really tricky bit. Could you add a comment?

Reply:
Done.